src-electron/zcl/zcl-loader-silabs.js

```diff
@@ -1751,6 +1807,252 @@ async function processDeviceTypes(db, filePath, packageId, data, context) {
+  return Promise.all(dataTypePromises)
+}
+
+/**
+ * Processes promises for loading individual tables per data type for
+ * atomics/baseline types.
+ *
+ * @param {*} db
+ * @param {*} filePath
+ * @param {*} packageId
+ * @param {*} knownPackages
+ * @param {*} toplevel
+ * @returns Promise of atomic/baseline processing.
+ */
+async function processAtomicTypes(
+  db,
+  filePath,
+  packageId,
+  knownPackages,
+  toplevel
+) {
+  let atomicPromises = []
+  if (dbEnum.zclType.atomic in toplevel) {
+    atomicPromises.push(
+      processNumber(db, filePath, packageId, knownPackages, toplevel.atomic)
+    )
+    atomicPromises.push(
+      processString(db, filePath, packageId, knownPackages, toplevel.atomic)
+    )
+    atomicPromises.push(
+      processEnumAtomic(db, filePath, packageId, knownPackages, toplevel.atomic)
+    )
+    atomicPromises.push(
+      processBitmapAtomic(
+        db,
+        filePath,
+        packageId,
+        knownPackages,
+        toplevel.atomic
+      )
+    )
+  }
+  return Promise.all(atomicPromises)
+}
+
+/**
+ * Processes promises for loading individual tables per data type for no-atomic
+ * and inherited types.
+ *
+ * @param {*} db
+ * @param {*} filePath
+ * @param {*} packageId
+ * @param {*} knownPackages
+ * @param {*} toplevel
+ * @param {*} featureClusters
+ * @returns Promise of non-atomic/inherited data type processing.
+ */
+async function processNonAtomicTypes(
+  db,
+  filePath,
+  packageId,
+  knownPackages,
+  toplevel,
+  featureClusters
+) {
+  let nonAtomicPromises = []
+  if (dbEnum.zclType.enum in toplevel) {
+    nonAtomicPromises.push(
+      processEnum(db, filePath, packageId, knownPackages, toplevel.enum)
+    )
+  }
+  if (dbEnum.zclType.bitmap in toplevel) {
+    nonAtomicPromises.push(
+      processBitmap(db, filePath, packageId, knownPackages, toplevel.bitmap)
+    )
+  }
+  // Treating features in a cluster as a bitmap
+  if (featureClusters && featureClusters.length > 0) {
+    featureClusters.forEach((fc) => {
+      nonAtomicPromises.push(
+        processBitmap(db, filePath, packageId, knownPackages, [
+          {
+            $: {
+              name: 'Feature',
+              type: 'BITMAP32',
+              cluster_code: [fc.code[0]]
+            }
+          }
+        ])
+      )
+    })
+  }
+  if (dbEnum.zclType.struct in toplevel) {
+    nonAtomicPromises.push(
+      processStruct(db, filePath, packageId, knownPackages, toplevel.struct)
+    )
+  }
+  return Promise.all(nonAtomicPromises)
+}
+
+/**
+ * Processes promises for loading items within a bitmap, struct, and enum data types.
+ *
+ * @param {*} db
+ * @param {*} filePath
+ * @param {*} packageId
+ * @param {*} knownPackages
+ * @param {*} toplevel
+ * @param {*} featureClusters
+ * @param {*} context
+ * @returns Promise of processing sub items within a bitmap, enum and structs.
+ */
+async function processSubItems(
+  db,
+  filePath,
+  packageId,
+  knownPackages,
+  toplevel,
+  featureClusters,
```

Comment:
alternatively, maybe set "featureClusters = []" by default in the function definition